### ISSUE TITLE
Inconsistent changes reported

### ISSUE THREAD

**Issue by watch24hrs-iiitd** [samuelcolvin/watchfiles#115]
Hi!
I encountered the following with `watchfiles 0.12`

**Error:**
Upon deletion changes, watchfiles is reporting addition changes.

**Code used:**
```python
async def main():
    async for changes in awatch("some dir"):
        for change_type, change_file in changes:
            print(change_type, change_file)
asyncio.run(main())
```

**Steps to reproduce:**
- Do a batch delete (multiselect files & then delete)
![image](https://user-images.githubusercontent.com/55681240/161764847-d5719277-b04e-45f9-a5fc-b2070320e2bd.png)
- The following changes are being reported
```
Change.added path-to-img1.bmp
Change.added path-to-img3.bmp
Change.deleted path-to-img0.bmp
Change.deleted path-to-img2.bmp
Change.deleted path-to-img4.bmp
```

**Comment by samuelcolvin** [comment]
Please provide: 
* What OS you're on. (Looks like ubuntu, but I can't be sure)
* What events are being printed with `async for changes in awatch("some dir", debug=True)`

Most likely, there lots of events triggered and watchfiles can't successfully clean them up - see:

https://github.com/samuelcolvin/watchfiles/blob/1ac9d3953fc209b9750247425a6c7ae02a0a6daa/src/lib.rs#L67-L77

We already remove "modified" events if a "deleted" or "added" event happens in the same batch, but I don't want to alwasy remove "added" events if a "deleted" occurred in case the file really was added and deleted in the same batch (which is common with short lived files like some of those generated by vim).

But with debug and OS info, we might be able to do slightly more filtering.

---

What's weird in this example is that `path-to-img1.bmp` is only added, which looks wrong, are you sure that's what you're seeing?

**Comment by watch24hrs-iiitd** [comment]
Hi!
Thanks for your reply.
Yes, I am on ubuntu (specifically 20.04.1).
I reran the experiment with `debug=True` and these are the following logs:
```
raw-event: Event { kind: Modify(Name(From)), paths: ["path-to-img0.bmp"], attr:tracker: Some(7750), attr:flag: None, attr:info: None, attr:source: None }
raw-event: Event { kind: Modify(Name(From)), paths: ["path-to-img1.bmp"], attr:tracker: Some(7752), attr:flag: None, attr:info: None, attr:source: None }
raw-event: Event { kind: Modify(Name(From)), paths: ["path-to-img2.bmp"], attr:tracker: Some(7754), attr:flag: None, attr:info: None, attr:source: None }
raw-event: Event { kind: Modify(Name(From)), paths: ["path-to-img3.bmp"], attr:tracker: Some(7756), attr:flag: None, attr:info: None, attr:source: None }
raw-event: Event { kind: Modify(Name(From)), paths: ["path-to-img4.bmp"], attr:tracker: Some(7758), attr:flag: None, attr:info: None, attr:source: None }
Change.deleted path-to-img2.bmp
Change.added path-to-img1.bmp
Change.added path-to-img3.bmp
Change.deleted path-to-img0.bmp
Change.deleted path-to-img4.bmp
```
And yes, I am only seeing the addition for `path-to-img1.bmp`, and interestingly the inconsistencies are kinda deterministic, and seem to always arise in  `path-to-img1.bmp` & `path-to-img3.bmp`.

**Comment by samuelcolvin** [comment]
Should be fixed in #119, which includes a brief explanation of the problem.

Note, you're not actually deleting files here, but moving them to the Rubbish bin, hence the `Modify(Name(From))`. If you did a proper delete with `Shift + Delete`, you'd get a `Remove(File)` event and this problem wouldn't happen.
